corehq/apps/reports/views.py

```diff
@@ -1082,7 +1082,7 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
         return "", []
 
     for config in configs:
-        content, excel_file = config.get_report_content(lang, attach_excel=attach_excel, couch_user=couch_user)
+        content, excel_file = _get_location_safe_report_content(config, domain, couch_user, lang, attach_excel)
```

Comment:
:nit

If functions with long list of args is not your favourite you can modify the logic like this

```
if is_location_restricted(domain, config):
    content, excel_file = "<content here>", []
else:
   content, excel_file = config.get_report_content(lang, attach_excel=attach_excel, couch_user=couch_user)

def is_location_restricted(domain, config):
    // return if condition is true
```

Reply:
https://github.com/dimagi/commcare-hq/pull/32964/commits/981a3ec8f8270adb2e8a33fcfa0f2a61b9242143